homeassistant/components/nmbs/sensor.py

```diff
@@ -114,11 +122,14 @@ def unique_id(self):
 
     @property
     def icon(self):
-        """Return the default icon or an alert icon if delays."""
-        if self._attrs and int(self._attrs["delay"]) > 0:
-            return DEFAULT_ICON_ALERT
+        """Return the default icon or an alert icon if delays or cancellations."""
+        if self._attrs:
+            delay = seconds_to_minutes(self._attrs["delay"])
+            departure_canceled = bool(int(self._attrs["canceled"]))
+            if delay > 0 or departure_canceled:
+                return "mdi:alert-octagon"
```

Comment:
A feature like that is supported by the `binary_sensor`. This should not be part of the sensor itself.